Samples/GettingStarted/Regression_TaxiFarePrediction/Program.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.IO;
+using Microsoft.ML.Models;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms;
+using Microsoft.ML;
+using System.Threading.Tasks;
+
+namespace Regression_TaxiFarePrediction
+{
+    internal static class Program
+    {
+        private static string AppPath => Path.GetDirectoryName(Environment.GetCommandLineArgs()[0]);
+        private static string TrainDataPath => Path.Combine(AppPath, @"..\..\..\..\datasets\", "taxi-fare-train.csv");
+        private static string TestDataPath => Path.Combine(AppPath, @"..\..\..\..\datasets\", "taxi-fare-test.csv");
+        private static string ModelPath => Path.Combine(AppPath,  "Models", "TaxiFareModel.zip");
+
+        private static async Task Main(string[] args)
+        {
+            var model = await TrainAsync();
+
+            Evaluate(model);
+
+            var prediction = model.Predict(TestTaxiTrips.Trip1);
+            Console.WriteLine($"Predicted fare: {prediction.FareAmount:0.####}, actual fare: 29.5");
+
+            Console.ReadLine();
+        }
+
+        private static async Task<PredictionModel<TaxiTrip, TaxiTripFarePrediction>> TrainAsync()
+        {
+            var pipeline = new LearningPipeline
+            {
+                new TextLoader<TaxiTrip>(TrainDataPath, useHeader: true, separator: ","),
+                new ColumnCopier(("FareAmount", "Label")),
+                new CategoricalOneHotVectorizer("VendorId",
+                    "RateCode",
+                    "PaymentType"),
+                new ColumnConcatenator("Features",
+                    "VendorId",
+                    "RateCode",
+                    "PassengerCount",
+                    "TripDistance",
+                    "PaymentType"),
+                new FastTreeRegressor()
+            };
+
+            Console.WriteLine("=============== Training model ===============");
+
+            var model = pipeline.Train<TaxiTrip, TaxiTripFarePrediction>();
+            
+            await model.WriteAsync(ModelPath);
+            
+            Console.WriteLine("=============== End training ===============");
+            Console.WriteLine("The model is saved to {0}", ModelPath);
+
+            return model;
+        }
+
+        private static void Evaluate(PredictionModel<TaxiTrip, TaxiTripFarePrediction> model)
+        {
+            var testData = new TextLoader<TaxiTrip>(TestDataPath, useHeader: true, separator: ",");
```

Comment:
>var testData = new TextLoader<TaxiTrip>(TestDataPath, useHeader: true, separator: ","); [](start = 12, length = 87)

Something is very wrong here... the loader is part of the model, or ought to be. We shouldn't have to respecify it, any more than we have to respecify the transforms.

Reply:
I was working on #5 and found out that Loader is actually not part of the trained model in ML.Net. I have opened another issue #216 for this blocker.